src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -161,7 +161,22 @@ export class RuntimeLanguageContext {
      * @param languageId : vscode language id or codewhisperer language name
      * @returns An object with a field language: CodewhispererLanguage, if no corresponding CodewhispererLanguage ID, plaintext is returned
      */
-    public getLanguageContext(languageId?: string): { language: CodewhispererLanguage } {
+    public getLanguageContext(languageId?: string, filename?: string): { language: CodewhispererLanguage } {
+        const extensionToLanguageMap: Record<string, CodewhispererLanguage> = {
+            tf: 'tf',
+            hcl: 'tf',
+            json: 'json',
+            yaml: 'yaml',
+            yml: 'yaml',
+            // Add more mappings if needed
```

Comment:
can we add tsx/jsx also?

Reply:
We can add `tsx`/`jsx` but the editor is not considering them as `plaintext`, If I am not wrong so if thats the case we do not see any difference even if we add them to this.

If editor recognizes `tsx` and `jsx` as `plaintext` then we can add them as `jsx`to`javascript` and `tsx` to `typescript`

Reply:
as Justin suggested, think making the method signature to be `getLanguageContext(document: vscode.TextDocument)` would be nicer and maybe rename it to be `getLanguageForDocument()`

Reply:
actually, can we also make this method returning `CodewhispererLanguage` instead of `{ language: CodewhispererLanguage }`, it's our tech debt and meaningless.